assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py

```diff
@@ -563,3 +565,87 @@ def _prepare_artifacts_dict(self) -> Dict:
             LLaVAMLflowLiterals.MODEL_DIR: os.path.join(self._model_dir, sd)
         }
         return artifacts_dict
+
+
+class SegmentAnythingMLFlowConvertor(PyFuncMLFLowConvertor):
+    """PyFunc MLfLow convertor for SAM models."""
+
+    MODEL_DIR = os.path.join(os.path.dirname(__file__), "segment_anything")
+    COMMON_DIR = os.path.join(os.path.dirname(os.path.dirname(__file__)), "common")
+
+    def __init__(self, **kwargs):
+        """Initialize MLflow convertor for SAM models."""
+        super().__init__(**kwargs)
+        if self._task != SupportedTasks.MASK_GENERATION.value:
+            raise Exception("Unsupported task")
+
+    def get_model_signature(self) -> ModelSignature:
+        """Return MLflow model signature with input and output schema for the given input task.
+
+        :return: MLflow model signature.
+        :rtype: mlflow.models.signature.ModelSignature
+        """
+        input_schema = Schema(
+            [
+                ColSpec(
+                    SegmentAnythingMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE_DATA_TYPE,
+                    SegmentAnythingMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE,
+                ),
+                ColSpec(
```

Comment:
No signature would still be more user friendly than specifying signature, where we'd have to give list as string

Reply:
@vivek-dani Yes, testing it without no-signature.

Reply:
Summary of testing with No-Signature:

Online Endpoint- With no signature, Json was read directly as pandas dataframe and endpoint was able to pass list of list of integers to the predict function as dataframe.

Batch Endpoint - In batch, with no signature case, csv file was read directly as pandas dataframe which expect string conversion of list of list of integers and predict functions gets a string in the case which is different from online endpoint.

Conclusion/Action- Decided to keep the backend logic simple and common for both batch and online endpoint, will take string as input for both and convert them to list of list of integers as of now.